lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1067,6 +1082,34 @@ private function executeInserts($class)
 
                 $this->addToIdentityMap($entity);
             }
+        } else {
+            foreach ($theseInsertions as $oid => $entity) {
```

Comment:
If the commit order calculator is working correctly, then this should simply be pushed down in the execution path, no?